tests/infrastructure/test_data_manager_stack.py

```diff
@@ -26,3 +32,850 @@ def test_sds_data_manager_stack(app, sds_id):
+                    "Arn",
+                ]
+            },
+            "BucketName": {"Ref": Match.string_like_regexp("DATABUCKET*")},
+            "NotificationConfiguration": {
+                "LambdaFunctionConfigurations": [
+                    {
+                        "Events": ["s3:ObjectCreated:*"],
+                        "LambdaFunctionArn": {
+                            "Fn::GetAtt": [
+                                Match.string_like_regexp("IndexerLambda*"),
+                                "Arn",
+                            ]
+                        },
+                    }
+                ]
+            },
+            "Managed": True,
+        },
+    )
+
+
+def test_secrets_manager_resource_count(template):
+    template.resource_count_is("AWS::SecretsManager::Secret", 1)
+
+
+def test_secrets_manager_resource_properties(template):
+    template.has_resource(
+        "AWS::SecretsManager::Secret",
+        {
+            "DeletionPolicy": "Delete",
+            "UpdateReplacePolicy": "Delete",
+        },
+    )
+
+
+def test_opensearch_domain_resource_count(template):
+    template.resource_count_is("AWS::OpenSearchService::Domain", 1)
+
+
+def test_opensearch_domain_resource_properties(template, sds_id):
+    template.has_resource_properties(
+        "AWS::OpenSearchService::Domain",
+        {
+            "DomainName": f"sdsmetadatadomain-{sds_id}",
+            "EngineVersion": "OpenSearch_1.3",
+            "ClusterConfig": {"InstanceType": "t3.small.search", "InstanceCount": 1},
+            "EBSOptions": {"EBSEnabled": True, "VolumeSize": 10, "VolumeType": "gp2"},
+            "NodeToNodeEncryptionOptions": {"Enabled": True},
+            "EncryptionAtRestOptions": {"Enabled": True},
+        },
+    )
+
+
+def test_custom_cloudwatch_log_resource_policy_count(template):
+    template.resource_count_is("Custom::CloudwatchLogResourcePolicy", 1)
+
+
+def test_custom_opensearch_access_policy_resource_count(template):
+    template.resource_count_is("Custom::OpenSearchAccessPolicy", 1)
+
+
+def test_custom_opensearch_access_policy_resource_properties(template):
+    template.has_resource(
+        "Custom::OpenSearchAccessPolicy",
+        {
+            "DeletionPolicy": "Delete",
+            "UpdateReplacePolicy": "Delete",
+        },
+    )
+
+    template.has_resource_properties(
+        "Custom::OpenSearchAccessPolicy",
+        {
+            "ServiceToken": {"Fn::GetAtt": [Match.string_like_regexp("AWS*"), "Arn"]},
+            "Create": {
+                "Fn::Join": [
+                    "",
+                    [
+                        '{"action":"updateDomainConfig","service":"OpenSearch","parameters":{"DomainName":"',
+                        {"Ref": Match.string_like_regexp("SDSMetadataDomain*")},
+                        '","AccessPolicies":"{\\"Statement\\":[{\\"Action\\":\\"es:*\\",\\"Effect\\":\\"Allow\\",\\"Principal\\":{\\"AWS\\":\\"*\\"},\\"Resource\\":\\"',
```

Comment:
I just saw this serialized JSON method, would that help for things like this with all the backslashes? https://docs.aws.amazon.com/cdk/api/v1/python/aws_cdk.assertions/README.html#serialized-json

Reply:
So far I haven't been able to use serialized_json() to get this to pass. The way this is broken up into serialized and deserialized json makes it so some serialized parts aren't valid json on their own, so I tried parsing this all into a dictionary and wrapping that with the serialized_json(), so I'm thinking I'm either missing something when I covert this all to a dictionary or because of the weird way this is broken up in the stack's json, the serialized_json() isn't going to work. I was trying to find why this is formatted like this (since other policy properties aren't) and haven't been able to find anything.

Reply:
OK, I wouldn't spend too much time on this. I was more curious if there was anything that stuck out to you as a way to make it look a little nicer. It is just a test and we can reformat it later if someone comes along with a better understanding of the differences.